### PR TITLE
TILA-1518: Code improvements and README rehaul

### PR DESCRIPTION
**Pull request by saaste**
## Change log
- Fixed small issues found by Sonar
- Rewrote README 🙈 

## Other notes
Since we discussed about the need for up-to-date and useful documentation, I spent some time trying to improve the README. Also, I went through code issues found by Sonar and tried to fix them. Hopefully this improves our reliability score.

## Deployment reminder
- No changes required

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,172 +1,205 @@
...
+## Environments
+The service is managed via Red Hat OpenShift. Management UIs:
+- [Development and Test environment](https://console-openshift-console.apps.arodevtest.hel.fi/topology/ns/hki-kanslia-aok-tilavarauspalvelu-test)
+- [Staging and Production environment](https://console-openshift-console.apps.platta.hel.fi/add/all-namespaces) (requires [Helsinki VPN](https://huolto.hel.fi/))
+
+The four environments are the following:
+- `dev`: runs the latests merged code
+- `test`: used for approval testing
+- `staging`: used to verify the release
+- `production`: what end-users are using
```
Comment (nicobav) [inline README.md:150]: What about something like:  _final product - for the end users_  ? 🤷‍♂️

File: README.md [hunk 2]
```diff
@@ -1,172 +1,205 @@
...
+When developing locally without Docker, you need to manually run the Celery worker
+by executing `celery -A tilavarauspalvelu worker --beat --loglevel info --scheduler django` in the project root if you want to run background jobs with Celery.
 
-In development environments it's easiest to use file system backend (the current default),
-you need to create a queue and processed folders and update env variables 
+In development environments, the file system backend is easiest to use (the current default).
+You need to create a queue and processed folders and update the environment variables 
 CELERY_QUEUE_FOLDER_OUT, CELERY_QUEUE_FOLDER_IN, CELERY_PROCESSED_FOLDER to match.
-Default value is ./broker/queue/ for in and out and ./broker/processed/ for processed. 
+The default values are ./broker/queue/ for in and out, and ./broker/processed/ for processed.
```
[the remark this answers is not in the document]
Comment (nicobav) [reply README.md:178]: this was just a pointer to ourselves or themselves for the future project members 😇

**Review by nicobav — COMMENTED** [review]
Looks nice, just a general initial comments and thoughts

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (saaste) [reply README.md:178]: Good idea! I could actually check this if I have extra time in this sprint 🔍

File: README.md [hunk 3]
```diff
@@ -242,3 +271,24 @@ urlpatterns += [path("__debug__/", include("debug_toolbar.urls"))]
...
+### Known setup issues
+
+#### Psycopg2 issue
+You might get an error when installing psycopg2. To fix the issue, you need to install OpenSSL and then update the LIBRARY_PATH env:
+```
+brew install openssl
+export LIBRARY_PATH=$LIBRARY_PATH:/usr/local/opt/openssl/lib/
+```
+
+#### OR-Tools issue
```
[the remark this answers is not in the document]
Comment (saaste) [reply README.md:287]: Same with this. I could spent some time on this if I have time left after doing the sprint tasks ☺️

**Review by nicobav — APPROVED** [review]
Nice 💪

**Pull request merged**